### PR TITLE
Add Mode Manager (Safe Mode)

### PR DESCRIPTION
**Pull request by yudataguy**
# Add ModeManager Component with Safe Mode

### Overview

This PR introduces a new **ModeManager** component that provides safe/normal/hibernation mode management for the PROVES satellite. Other components can call on the mode and make changes to the proves when necessary. This is the reduced version due to auto safe mode not passing tests, will implement this once the manager is stable. 

⚠️ I'm afraid there might be some specification issue due to the understanding of the chart. Hibernation mode will come at a later time once this is implemented. Please let me know if any part need to be modified. 

### Key Features

When in safe mode, load switches etc is being turned off to preserve power.

## Related Issues/Tickets

#106 

## How Has This Been Tested?

- [ ] Unit tests
- [x] Integration tests
- [ ] Z Tests
- [ ] Manual testing (describe steps)

## Checklist

- [ ] Written detailed sdd with requirements, channels, ports, commands, telemetry defined and correctly formatted and spelled
- [ ] Have written relevant integration tests and have documented them in the sdd
- [ ] Have done a code review with
- [ ] Have tested this PR on every supported board with correct board definitions

## Further Notes / Considerations

Input Ports:

- watchdogFaultSignal - Receives watchdog fault notifications from Watchdog component
- forceSafeMode - Allows other components to trigger safe mode
- clearAllFaults - Automated fault recovery (use with caution)

Commands:

- FORCE_SAFE_MODE - Forces immediate safe mode entry
- EXIT_SAFE_MODE - Exits safe mode (requires all faults cleared)

```
┌─────────────┐
│   NORMAL    │◄────────────────────────┐
│   MODE      │                         │
└─────────────┘                         │
      │                                 │
      │ • SAFE_MODE command             │
      │                                 │
      ▼                                 │
┌─────────────┐                         │
│  SAFE MODE  │─────────────────────────┘
│             │
│ Actions:    │   Recovery Requirements:
│ • Turn OFF  │  
│   8 loads   │   
│ • Persist   │   
│   to file   │   4. EXIT_SAFE_MODE (validates all cleared)
│ • Log event │
└─────────────┘
```

### CODE REVIEW

File: FprimeZephyrReference/Components/ModeManager/ModeManager.fpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        @ Port to force safe mode entry (callable by other components)
+        async input port forceSafeMode: Fw.Signal
+
+
+        # ----------------------------------------------------------------------
+        # Output Ports
+        # ----------------------------------------------------------------------
+
+        @ Port to notify other components of mode changes
+        output port modeChanged: Fw.Signal
```
Comment (nateinaction) [inline FprimeZephyrReference/Components/ModeManager/ModeManager.fpp]: It looks like a `modeChanged` output port wouldn't be enough for other components to react. I don't think the `Fw.Signal` carries enough information to say what mode the mode manager is in so the down stream components would only know that a mode change occurred. I think you will want either to:
- add an input port so that downstream components can check to see what mode the satellite is in (my vote)
- or you'll want the `modeChanged` output port to return a datatype that carries enough information so downstream components know what mode the satellite has changed into.
- or you'll want who "mode changed" output ports, one for safe mode and another for nominal mode (this is my least favorite implementation)

What are your thoughts? Am I missing something about the implementation?

File: FprimeZephyrReference/Components/ModeManager/docs/sdd.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
+# Components::ModeManager
+
+The ModeManager component manages system operational modes and orchestrates safe mode transitions based on voltage monitoring, watchdog faults, and communication timeouts. It provides centralized mode management for the satellite, controlling power to non-critical components during safe mode and maintaining mode state across reboots.
+
+## Requirements
+| Name | Description | Validation |
+|---|---|---|
+| MM0001 | The ModeManager shall maintain two distinct operational modes: NORMAL and SAFE_MODE | Integration Testing |
```
Comment (ineskhou) [inline FprimeZephyrReference/Components/ModeManager/docs/sdd.md]: we also want payload mode, and eventually hibernate modes -- i assume ou'dd add those later but should probably be in the sdd

File: FprimeZephyrReference/Components/ModeManager/docs/sdd.md [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+|---|---|---|
+| test_01_initial_telemetry | Verifies initial telemetry can be read | Basic functionality |
+| test_04_force_safe_mode_command | Tests FORCE_SAFE_MODE command enters safe mode | Safe mode entry |
+| test_05_safe_mode_increments_counter | Verifies SafeModeEntryCount increments | State tracking |
+| test_06_safe_mode_turns_off_load_switches | Verifies all load switches turn off in safe mode | Power management |
+| test_07_safe_mode_emits_event | Verifies EnteringSafeMode event with correct reason | Event emission |
+| test_13_exit_safe_mode_fails_not_in_safe_mode | Tests EXIT_SAFE_MODE validation | Error handling |
+| test_14_exit_safe_mode_success | Tests successful safe mode exit | Safe mode exit |
+| test_18_force_safe_mode_idempotent | Tests FORCE_SAFE_MODE is idempotent | Edge cases |
+| test_19_safe_mode_state_persists | Verifies mode persistence to flash | State persistence |
```
Comment (Mikefly123) [inline FprimeZephyrReference/Components/ModeManager/docs/sdd.md]: Is there a particular reason that the test numbers are not sequential?

**Review by ineskhou — APPROVED** [review]
LGTM

**Review by Mikefly123 — APPROVED** [review]
LGTM! Integration tests are passing
